packages/material-ui-system/src/styleFunctionSx.js

```diff
@@ -45,32 +53,40 @@ function styleFunctionSx(props) {
     return styles;
   }
 
-  let css = {};
+  const emptyBreakpoints = createEmptyBreakpointObject(theme.breakpoints);
+  const breakpointsKeys = Object.keys(emptyBreakpoints)
+
+  let css = emptyBreakpoints;
 
   Object.keys(styles).forEach((styleKey) => {
     if (typeof styles[styleKey] === 'object') {
       if (filterProps.indexOf(styleKey) !== -1) {
+        // breakpoints, need deepmerge
         css = deepmerge(css, getThemeValue(styleKey, styles[styleKey], theme));
       } else {
         const breakpointsValues = handleBreakpoints({ theme }, styles[styleKey], (x) => ({
           [styleKey]: x,
         }));
 
         if (objectsHaveSameKeys(breakpointsValues, styles[styleKey])) {
-          const transformedValue = styleFunctionSx({ sx: styles[styleKey], theme });
-          css[styleKey] = transformedValue;
+          css[styleKey] = styleFunctionSx({ sx: styles[styleKey], theme });
         } else {
+          // breakpoints for regular CSS values, need deep merge
           css = deepmerge(css, breakpointsValues);
         }
       }
     } else if (typeof styles[styleKey] === 'function') {
       css = deepmerge(css, { [styleKey]: styles[styleKey](theme) });
     } else {
-      css = deepmerge(css, getThemeValue(styleKey, styles[styleKey], theme));
+      // simple value no need for deepmerge
+      const result = getThemeValue(styleKey, styles[styleKey], theme);
+      Object.keys(result).forEach((key) => {
+        css[key] = result[key];
+      });
     }
   });
 
-  return mergeBreakpointsInOrder(theme.breakpoints, css);
```

Comment:
We do not really need to merge the breakpoints with `css`, we can just initialize `css` with the empty breakpoints and in the end, remove the unused ones.